### PR TITLE
AVX-54473: Add support for k8s smart groups

### PR DESCRIPTION
**Pull request by rpanzer-aviatrix**
This PR extends the Aviatrix Provider so that Smart Groups can be built from artefacts in Kubernetes clusters.

### CODE REVIEW

File: docs/resources/aviatrix_smart_group.md [hunk 1]
```diff
@@ -53,13 +67,23 @@ The following arguments are supported:
...
+      This property can only be used when `type` is set to "k8s".
+    * `k8s_namespace` - (Optional) - Kubernetes namespace this expression matches.
+      This property can only be used when `type` is set to "k8s".
+    * `k8s_service` - (Optional) - Kubernetes service name this expression matches.
+      This property can only be used when `type` is set to "k8s".
+      This property must not be used when `k8s_pod` is set.
+    * `k8s_pod` - (Optional) - Kubernetes pod name this expression matches. 
+      This property can only be used when `type` is set to "k8s".
+      This property must not be used when `k8s_service` is set.
     * `tags` - (Optional) - Map of tags this expression matches.
```
Comment (CyrusJavan) [inline docs/resources/aviatrix_smart_group.md:87]: More accurate?
```suggestion
    * `tags` - (Optional) - Map of cloud resource tags this expression matches or kubernetes labels when type is "k8s".
```

Comment (rpanzer-aviatrix) [reply docs/resources/aviatrix_smart_group.md:87]: At this point tags are not allows with the type "k8s" :D

File: aviatrix/resource_aviatrix_smart_group.go [hunk 2]
```diff
@@ -3,13 +3,18 @@ package aviatrix
...
+	"regexp"
 
 	"github.com/AviatrixSystems/terraform-provider-aviatrix/v3/goaviatrix"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
 )
 
+var (
+	k8sNameRe = regexp.MustCompile(`^[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?$`)
```
Comment (dnardo) [inline aviatrix/resource_aviatrix_smart_group.go]: Might be nice to have a comment like
// K8s resource names must adhere to rfc1123 naming conventions.

**Review by dnardo — DISMISSED** [review]

File: aviatrix/resource_aviatrix_smart_group.go (hunk 2 above)
Comment (rpanzer-aviatrix) [reply aviatrix/resource_aviatrix_smart_group.go]: Renamed the variable to mention the rfc and added a comment.

**Review by dnardo — APPROVED** [review]

**Pull request merged**